lib/pnet.go

```diff
@@ -0,0 +1,42 @@
+package vole
+
+import (
+	"context"
+	"fmt"
+	"os"
+
+	"github.com/libp2p/go-libp2p/core/pnet"
+)
+
+type pnetPSKContextKey struct{}
+
+func WithPnetPSK(ctx context.Context, psk pnet.PSK) context.Context {
```

Comment:
Note: I went with the context plumbing approach mostly for ease. I'm also happy to swap this to something else (e.g. passing around libp2p options or switching to passing a host explicitly and then exposing a default constructor).